File: Liczby_Zespolone.cpp
#include <iostream>
#include <cmath>
using namespace std;

class ZESPOLONA
{
    public:
    double x, y;
    ZESPOLONA(double a = 0, double b = 0) : x(a), y(b) {}
    ZESPOLONA dodaj(ZESPOLONA a);
    ZESPOLONA odejmij(ZESPOLONA a);
    ZESPOLONA pomnoz(ZESPOLONA a);
    ZESPOLONA podziel(ZESPOLONA a);
    void wyswietl(void);
};
ZESPOLONA ZESPOLONA::dodaj(ZESPOLONA a)
{
    ZESPOLONA wynik;
    wynik.x = x + a.x;
    wynik.y = y + a.y;
    return wynik;
}
ZESPOLONA ZESPOLONA::odejmij(ZESPOLONA a)
{
    ZESPOLONA wynik;
    wynik.x = x - a.x;
    wynik.y = y - a.y;
    return wynik;
}
ZESPOLONA ZESPOLONA::pomnoz(ZESPOLONA a)
{
    ZESPOLONA wynik;
    wynik.x = x * a.x - y * a.y;
    wynik.y = y * a.x + y * a.y;
    return wynik;
}
ZESPOLONA ZESPOLONA::podziel(ZESPOLONA a)
{
    ZESPOLONA wynik;
    wynik.x = (x * a.x + y * a.y)/(pow(a.x, 2) + pow(a.y, 2));
    wynik.y = (y * a.x - y * a.y)/(pow(a.x, 2) + pow(a.y, 2));
    return wynik;
}
void ZESPOLONA::wyswietl(void)
{
    cout << "Czesc rzeczywista: " << x << ", "
    << "czesc urojona: " << y << endl;
}
int main()
{
    //Przykladowe testowanie programu...
    ZESPOLONA pierwsza_liczba(2,5); // 2 + 5i
    ZESPOLONA druga_liczba(4,2); //4 + 2i
    ZESPOLONA wynik;
    wynik = pierwsza_liczba.dodaj(druga_liczba);
    wynik.wyswietl();
    return 0;
}

File: Zadanie2/LiczbyZespolone/lib/LiczbyZespolone.rb
require_relative "LiczbyZespolone/version"

module LiczbyZespolone
  class ZESPOLONA
	attr_accessor :x, :y  
	def initialize(x = 0, y = 0)
 		@x = x
 		@y = y
 	end
 	
	def wyswietl()
		puts "czesc rzeczywista = #{@x}, czesc urojona = #{@y}\n" 
	end
	
	def dodaj(a)
		c = ZESPOLONA.new  	
		c.x = @x + a.x
		c.y = @y + a.y
		return c
 	end
	def odejmij(a)
		c = ZESPOLONA.new  	
		c.x = @x - a.x
		c.y = @y - a.y
		return c
 	end
	def pomnoz(a)
		c = ZESPOLONA.new  	
		c.x = @x * a.x - @y * a.y
		c.y = @y * a.x + @x * a.y
		return c
 	end
	def podziel(a)
		c = ZESPOLONA.new  	
		c.x = (@x * a.x + @y * a.y)/((a.x ** 2 + a.y ** 2)*1.0)
		c.y = (@y * a.x - @x * a.y)/((a.x ** 2 + a.y ** 2)*1.0)
		return c
 	end
  end
end

